src/tezWrapper.ml

```diff
@@ -0,0 +1,279 @@
+
+let[@inline] transfer (state: tez_wrapper_state) (xs: fa2_transfer list) : LigoOp.operation list * tez_wrapper_state =
+  (* TODO: fa2_transfers always come in a list, so it's not very easy to emit
+   * all the additional operations for moving tez around (from vault to
+   * vault) at the same time.  Either we should do a second pass to transfer
+   * the tez amounts, or we have to change our current abstractions. *)
+  let _ = ensure_no_tez_given () in
+  let state, ops = tez_wrapper_fa2_run_transfer (state, xs) in (* TODO: Uses specialized tez_wrapper_fa2_run_transfer *)
+  (ops, state)
+
+let[@inline] update_operators (state: tez_wrapper_state) (xs: fa2_update_operator list) : LigoOp.operation list * tez_wrapper_state =
+  let _ = ensure_no_tez_given () in
+  let state = { state with fa2_state = fa2_run_update_operators (state.fa2_state, xs) } in (* NOTE: No need for specialized calls, since the spec does not require checking the token_ids here. *)
+  (([]: LigoOp.operation list), state) (* no need to originate vaults *)
+
+(*****************************************************************************)
+(**                      {1 WRAPPER ENTRYPOINTS}                             *)
+(*****************************************************************************)
+
+let[@inline] deposit (state: tez_wrapper_state) (_: unit) : LigoOp.operation list * tez_wrapper_state =
+  (* 1. Update the balance on the ledger *)
+  let state_fa2_state = ledger_issue_tez_token (state.fa2_state, !Ligo.Tezos.sender, !Ligo.Tezos.amount) in
+  let state = { state with fa2_state = state_fa2_state } in
+  (* 2. Create a vault, if it does not exist already, and update the map. *)
+  let op_opt, state, vault_address = find_vault_address state !Ligo.Tezos.sender in
+  (* 3. Transfer the actual tez to the vault of the sender. *)
+  let op = match (LigoOp.Tezos.get_entrypoint_opt "%vault_receive_tez" vault_address : unit Ligo.contract option) with
+    | Some c -> LigoOp.Tezos.unit_transaction () !Ligo.Tezos.amount c
+    | None -> (Ligo.failwith error_GetEntrypointOptFailureVaultReceiveTez : LigoOp.operation) in
+  match op_opt with
+  | None -> ([op], state)
+  | Some origination -> ([origination; op], state)
+
+let[@inline] withdraw (state: tez_wrapper_state) (amnt: Ligo.tez) : LigoOp.operation list * tez_wrapper_state =
+  (* 1. Ensure no tez given *)
+  let _ = ensure_no_tez_given () in
+  (* 2. Reduce the balance of the tez owner *)
+  let state_fa2_state = ledger_withdraw_tez_token (state.fa2_state, !Ligo.Tezos.sender, amnt) in
+  let state = { state with fa2_state = state_fa2_state; } in
+  (* 3. Create a vault, if it does not exist already, and update the map. *)
+  let op_opt, state, vault_address = find_vault_address state !Ligo.Tezos.sender in
+  (* 4. Instruct the vault to send the actual tez to the owner *)
+  let op = match (LigoOp.Tezos.get_entrypoint_opt "%vault_send_tez_to_contract" vault_address : (Ligo.tez * Ligo.address) Ligo.contract option) with
```

Comment:
I'm wondering if we want to send the tez to a contract or via a bare transaction here to enable withdrawals directly to user accounts? Not sure if there are any negative implications to doing this though